File: backend/app/schemas/dataset.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any, List
from datetime import datetime


class DatasetBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    modality: str = Field(..., pattern="^(text|image|audio|video|multimodal)$")
    size_bytes: int = Field(..., gt=0)
    file_count: int = Field(..., gt=0)
    description: Optional[str] = None
    storage_path: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class DatasetCreate(DatasetBase):
    pass


class DatasetResponse(DatasetBase):
    id: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    stats: Optional[Dict[str, Any]] = None

    class Config:
        from_attributes = True


class DatasetListResponse(BaseModel):
    data: List[DatasetResponse]
    pagination: Dict[str, int]

File: frontend/src/types/dataset.ts
export type Modality = 'text' | 'image' | 'audio' | 'video' | 'multimodal';

export interface Dataset {
  id: string;
  name: string;
  modality: Modality;
  size_bytes: number;
  file_count: number;
  description?: string;
  storage_path?: string;
  created_at: string;
  updated_at?: string;
  metadata?: Record<string, any>;
  stats?: {
    experiments_using?: number;
    last_used?: string;
  };
}

export interface DatasetFilters {
  modality?: Modality;
  search?: string;
}

export interface PaginatedResponse<T> {
  data: T[];
  pagination: {
    total: number;
    page: number;
    page_size: number;
    total_pages: number;
  };
}
